Sources/OpenCombine/Publishers/Publishers.DropWhile.swift

```diff
@@ -46,88 +45,107 @@ extension Publishers {
         public func receive<SubscriberType: Subscriber>(subscriber: SubscriberType)
             where Output == SubscriberType.Input, SubscriberType.Failure == Error
         {
-            let dropWhile = _DropWhile<Upstream, SubscriberType, (Output) throws -> Bool>(
-                downstream: subscriber, predicate: predicate
-            )
-
-            upstream.receive(subscriber: dropWhile)
+            let inner = Inner<SubscriberType>(downstream: subscriber,
+                                              predicate: predicate)
+            upstream.receive(subscriber: inner)
         }
     }
 }
 
-private final class _DropWhile<Upstream: Publisher, Downstream: Subscriber, Predicate>
-    : Subscriber,
+private class _DropWhile<Upstream: Publisher, Downstream: Subscriber>
+    : OperatorSubscription<Downstream>,
       CustomStringConvertible,
-      CustomReflectable,
       Subscription
-          where Upstream.Output == Downstream.Input
+    where Upstream.Output == Downstream.Input
 {
-
-    typealias Input = Downstream.Input
+    typealias Input = Upstream.Output
     typealias Failure = Upstream.Failure
+    typealias GenericPredicate = (Input) -> Result<Bool, Downstream.Failure>
```

Comment:
Maybe something like `DropPredicate` would be a more clear name?

Reply:
IMO adding the "Drop" word won't bring any new information to the reader — we're already dealing with DropWhile.

Reply:
That's fair enough.  I think I'm just finding the wording `Generic` confusing because I'm expecting it to be more encompassing than a local typealias.